File: src/src/main/java/br/com/blz/testjava/models/Warehouse.java
package br.com.blz.testjava.models;

import com.fasterxml.jackson.annotation.JsonIgnore;


import lombok.Data;
import org.hibernate.validator.constraints.NotEmpty;

import javax.persistence.*;
import javax.validation.constraints.NotNull;


@Data
@Entity
public class Warehouse {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonIgnore
    private long id;

    @NotEmpty(message = "locality can't be empty")
    @NotNull(message = "locality is required")
    private String locality;

    @NotNull(message = "quantity is required")
    private Integer quantity;

    @NotEmpty(message = "type can't be empty")
    @NotNull(message = "type is required")
    private String type;
}

File: src/main/kotlin/br/com/blz/testjava/model/Warehouse.kt
package br.com.blz.testjava.model

import com.fasterxml.jackson.annotation.JsonProperty

data class Warehouse(
  @JsonProperty("locality") val locality: String = "",
  @JsonProperty("quantity") val quantity: Int = 0,
  @JsonProperty("type") val type: String = ""
)
